src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -580,22 +578,11 @@ private Set<ProgrammingExerciseTestCase> filterTestCasesForStudents(final Set<Pr
      */
     private Result calculateScoreForResult(Set<ProgrammingExerciseTestCase> testCases, Set<ProgrammingExerciseTestCase> testCasesForCurrentDate, @NotNull Result result,
             ProgrammingExercise exercise, boolean applySubmissionPolicy) {
-        List<Feedback> testCaseFeedback = new ArrayList<>();
-        List<Feedback> staticCodeAnalysisFeedback = new ArrayList<>();
-        for (var feedback : result.getFeedbacks()) {
-            if (feedback.getType() != FeedbackType.AUTOMATIC) {
-                continue;
-            }
+        Map<Boolean, List<Feedback>> feedbacksGroupedByType = result.getFeedbacks().stream().collect(Collectors.groupingBy(Feedback::isStaticCodeAnalysisFeedback));
```

Comment:
```suggestion
        Map<Boolean, List<Feedback>> feedbacksGroupedByType = result.getFeedbacks().stream().filter(feedback -> FeedbackType.AUTOMATIC.equals(feedback.getType())).collect(Collectors.groupingBy(Feedback::isStaticCodeAnalysisFeedback));
```

The old logic had a filter for automatic feedbacks. This is important as otherwise manual feedbacks are considered as test case feedback, e.g., after the weights for the grading have been changed and a score recalculation is triggered by the instructor.